### PR TITLE
Adopt sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**

### CODE REVIEW

**Comment by armanbilge** [comment]
I don't understand why CI is hanging on sbt startup :(

File: build.sbt
Comment (rossabaker) [inline build.sbt:16]: Why?  Are there warnings?

**Review by rossabaker — CHANGES_REQUESTED** [review]
This should have a `startYear` of `2012`.  Otherwise, looking good.

File: build.sbt
Comment (armanbilge) [reply build.sbt:16]: Yep, lots of them in the CI logs.

Comment (rossabaker) [reply build.sbt:17]: Alright, one step at a time.

**Review by rossabaker — APPROVED** [review]

**Comment by armanbilge** [comment]
Btw, since this one wasn't set up for CI release previously I'm assuming there are no secrets installed. This could be a chance to try out GitHub org-level secrets.

**Comment by rossabaker** [comment]
What precisely do we need these days?  A sonatype username and password and a GPG passphrase?

**Comment by armanbilge** [comment]
Sonatype username.password and base64 GPG private key that is not passphrase-protected.

**Comment by rossabaker** [comment]
The passphrase is easy.  I'd like to create Sonatype credentials for a Typelevel bot account that can only publish to Typelevel.  In case my credentials get exfilitrated, I don't want the blast radius any wider than this.

**Comment by rossabaker** [comment]
@typelevel/steering Can someone help with https://github.com/orgs/typelevel/teams/maintainers/discussions/3 so we can set up that Sonatype account?  I got kicked off the Discord server and lost all my roles again, so I can't discuss on channel.

**Comment by rossabaker** [comment]
I have added the private key for 0x598648A65DF64506 to the org as PGP_SECRET.  

To unblock things, I set up a `SONATYPE_USERNAME` and `SONATYPE_PASSWORD` in this project only.

**Pull request merged**
